app/config_tools.py

```diff
@@ -138,85 +190,102 @@ def __init__(self, config_path):
+            self.client_id = check_for_attribute(config, "client_id", parent="trakt")
+            self.client_secret = check_for_attribute(config, "client_secret", parent="trakt")
             self.authorization = config['authorization']
             Trakt.configuration.defaults.client(self.client_id, self.client_secret)
             Trakt.configuration.defaults.oauth.from_response(self.authorization)
         elif TraktClient.valid == None:
             print("| Connecting to trakt...")
-            try:
-                self.client_id = check_for_attribute(config, "client_id", text="trakt sub-attribute {}")
-                self.client_secret = check_for_attribute(config, "client_secret", text="trakt sub-attribute {}")
-                self.authorization = config['authorization']
+            message = ""
+            try:                        self.client_id = check_for_attribute(config, "client_id", parent="trakt", throw=True)
+            except SystemExit as e:     message = message + "\n" + str(e) if len(message) > 0 else str(e)
+            try:                        self.client_secret = check_for_attribute(config, "client_secret", parent="trakt", throw=True)
+            except SystemExit as e:     message = message + "\n" + str(e) if len(message) > 0 else str(e)
+            if len(message) > 0:
+                print(message)
+                TraktClient.valid = False
+            else:
+                try:
+                    if 'authorization' in config and config['authorization']:
+                        self.authorization = config['authorization']
+                    else:
+                        self.authorization = {'access_token': None, 'token_type': None, 'expires_in': None, 'refresh_token': None, 'scope': None, 'created_at': None}
 
-                def check_trakt (auth):
-                    try:
-                        #TODO: Validate Trakt Connection
-                        return True
-                    except:
-                        return False
+                    Trakt.configuration.defaults.client(self.client_id, self.client_secret)
 
-                if not check_trakt(self.authorization):
-                    self.authorization = {'access_token': None, 'token_type': None, 'expires_in': None, 'refresh_token': None, 'scope': None, 'created_at': None}
-                    print("| Stored Authorization Failed")
-                Trakt.configuration.defaults.client(self.client_id, self.client_secret)
-                self.updated_authorization = trakt_helpers.authenticate(self.authorization, headless=Config.headless)
+                    def check_trakt (auth):
+                        try:
+                            Trakt.configuration.defaults.oauth.from_response(auth)
+                            trakt_list_path = urlparse("https://trakt.tv/users/movistapp/lists/christmas-movies").path
+                            trakt_list_items = trakt.Trakt[trakt_list_path].items()
```

Comment:
This seems like a duplicate check with what is in `trakt_helpers.py`? @mza921 you know this code better than I do

https://github.com/mza921/Plex-Auto-Collections/blob/dda8dc9374b18103a5ee7c3d689e1afda080ffcd/app/trakt_helpers.py#L12-L17

It seems like it would be better to consolidate with one check with proper error handling. I like using `Trakt['users/settings']` instead of a list, personally, for a variety of reasons (this list could change, the list is publicly accessible / you don't need to be authenticated so could produce false positives, etc.).

Reply:
as far as i can tell the line below doesn't check much

     with Trakt.configuration.oauth.from_response(authorization, refresh=True): 
         if Trakt['users/settings']: 

to test this i added the two lines below right between
                    `Trakt.configuration.defaults.client(self.client_id, self.client_secret)`
and
                    `def check_trakt (auth):`


                    Trakt.configuration.defaults.client(self.client_id, self.client_secret)
                    with Trakt.configuration.oauth.from_response(self.authorization, refresh=True):
                        print("Pass" if Trakt['users/settings'] else "Fail")
                    def check_trakt (auth):

With that line in there no matter what i do to the lines in the config either a check i have before this flags it (blank attributes) or this will pass. I cant get this to fail at all so I don't know how I could use it to authenticate.

with my config as this

trakt:
    client_id: q
    client_secret: q
    authorization:
        access_token:
        token_type:
        expires_in:
        refresh_token:
        scope:
        created_at:

When I run the program with my test addition i still get this output saying it passed the original test before failing my new one.

|===================================================================================================|
| Connecting to trakt...
Pass
| Stored Authorization Failed
| Navigate to: https://trakt.tv/oauth/authorize?client_id=q&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code
| If you get an OAuth error your client_id or client_secret is invalid
| trakt pin (case insensitive):

So all in all I agree using a list that could possibly be deleted isn't the best idea but while it is publicly available you still have to have proper authentication to get the list using the Trakt python class so I don't think we would have any false positives in relation to that. One solution to the first problem is that one of us makes a public list that we won't delete for the program to check against instead of the one I was using. I'm definitely open to doing the authentication in a different way but following how it would fail before is how i got to this method of authenticating the credentials.

Reply:
I'm also going to add in a message so that if the user needs to authenticate while in headless mode it will print out a message saying as much fixing the error [here](https://github.com/mza921/Plex-Auto-Collections/issues/55#issuecomment-715524467) but I wanna see what we're gonna decide on all this above first

Reply:
I think it should be changed to:
```if Trakt['users/settings'].get():```

Reply:
and you would be correct I updated the authorization it should be good now